### PR TITLE
Introduce fields in schema to describe synthetic data

### LINKED ISSUE #62: Metadata fields related to synthetic data
Track which simulation code was used, etc

### PR DESCRIPTION
**Pull request by WardLT**
Our initial schemas focused only on data from battery cycling experiments. This PR is a first step towards expanding the types of data and adds fields associated with describing modeling data.

While at it, I'm starting to map the fields I create to the BattINFO ontology (see #30)

Fixes #62 

To do list:
- [ ] Check with @jsimonclark that our approach to mapping schema fields isn't terrible. We we can go through further refinements later
- [x] Capture software name, type and version number

### CODE REVIEW

**Comment by WardLT** [comment]
@jsimonclark, if you (or someone else from BattINFO) do get a chance to look at this, here are the two parts of relevance:

- A high-level description of how use use BattINFO: https://github.com/ROVI-org/battery-data-toolkit/blob/synth-schema/docs/schemas.md
- Our python implementation for using the ontology to generate fields names, which I'm going to use when rendering documentation - https://github.com/ROVI-org/battery-data-toolkit/blob/synth-schema/batdata/schemas/ontology.py

**Comment by jsimonclark** [comment]
@WardLT, this is looking really neat!

It looks like you are using BattINFO as a source for things like labels, elucidations, and as a "see also" reference for some of your fields. The mapping is done by adding an `iri` field to the class description, like in how you treated [nominal_capacity.](https://github.com/ROVI-org/battery-data-toolkit/blob/b22a341d9bf2063912a7308a9caf746b564ba039/batdata/schemas/battery.py#L53). Is that a correct summary? 

This is a good approach to start integrating BattINFO terms. The real power of RDF comes when you start making graphs, which opens things up for semantic querying via SPARQL, machine-readable linking with other sources, and easier automation. JSON-LD is usually my recommended way for doing that, because it is easy to understand and can build on existing JSON infrastructure. Let me know if you'd like to discuss and we can setup a call.

**Comment by jsimonclark** [comment]
Also, for bibliographic and general metadata, keep resources like [schema.org](https://schema.org/) and [dcterms](https://www.dublincore.org/specifications/dublin-core/dcmi-terms/) in mind. I'm especially a fan of schema.org because it strikes a good balance of simplicity and features - and it helps with findability in Google or other search engines. For example, [here ](https://github.com/ROVI-org/battery-data-toolkit/blob/b22a341d9bf2063912a7308a9caf746b564ba039/batdata/schemas/battery.py#L42) you could map to [schema:manufacturer](https://schema.org/manufacturer).

File: batdata/schemas/modeling.py
Comment (victorventuri) [inline batdata/schemas/modeling.py:1]: I don't think I understand what this is used for...

Comment (victorventuri) [reply batdata/schemas/modeling.py:1]: I think I get it now, oops...

**Comment by WardLT** [comment]
Thanks for the feedback @jsimonclark ! You've got my goals right.

I'll see if I can link to dcterms while I continue fleshing out ontology support.

I will probably need to talk to you more about going the next step in going towards better RDF/JSON-LD support. That's a new technology stack to me and I would be glad to learn from you!

**Pull request merged**